fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraVersioningIT.java

```diff
@@ -406,24 +408,49 @@ public void testGetTimeMapResponseForBinaryDescription() throws Exception {
     }
 
     private void verifyTimemapResponse(final String uri, final String id) throws Exception {
-        final List<Link> listLinks = new ArrayList<Link>();
+        verifyTimemapResponse(uri, id, null);
+    }
+
+    private void verifyTimemapResponse(final String uri, final String id, final String mementoDateTime)
+        throws Exception {
+        final String ldpcvUri = uri + "/" + FCR_VERSIONS;
+        final List<Link> listLinks = new ArrayList<>();
         listLinks.add(Link.fromUri(uri).rel("original").build());
         listLinks.add(Link.fromUri(uri).rel("timegate").build());
         listLinks
-                .add(Link.fromUri(uri + "/" + FCR_VERSIONS).rel("self").type(APPLICATION_LINK_FORMAT)
+                .add(Link.fromUri(ldpcvUri).rel("self").type(APPLICATION_LINK_FORMAT)
                         .build());
-        final Link[] expectedLinks = listLinks.toArray(new Link[3]);
+        if (mementoDateTime != null) {
+            final TemporalAccessor instant = RFC_1123_DATE_TIME.parse(mementoDateTime);
+            listLinks.add(Link.fromUri(ldpcvUri + "/" + MEMENTO_DATETIME_ID_FORMATTER.format(instant))
+                              .rel("memento")
+                              .param("datetime", mementoDateTime)
+                              .build());
+        }
 
+        final Link[] expectedLinks = listLinks.toArray(new Link[0]);
         final HttpGet httpGet = getObjMethod(id + "/" + FCR_VERSIONS);
         httpGet.setHeader("Accept", APPLICATION_LINK_FORMAT);
         try (final CloseableHttpResponse response = execute(httpGet)) {
             assertEquals("Didn't get a OK response!", OK.getStatusCode(), getStatus(response));
             checkForLinkHeader(response, uri, "original");
             checkForLinkHeader(response, VERSIONING_TIMEMAP_TYPE, "type");
-            final List<String> bodyList = Arrays.asList(EntityUtils.toString(response.getEntity()).split(","));
+            final List<String> bodyList = Arrays.asList(EntityUtils.toString(response.getEntity()).split(",\n"));
+            //the links from the body are not
             final Link[] bodyLinks = bodyList.stream().map(String::trim).filter(t -> !t.isEmpty())
-                    .map(Link::valueOf).toArray(Link[]::new);
+                                                      .map(Link::valueOf).toArray(Link[]::new);
```

Comment:
You can do the sort right on the stream if you want. Something like this should work.
```
final Link[] bodyLinks = bodyList.stream().map(String::trim)
.filter(t -> !t.isEmpty()).map(Link::valueOf)
.sort((a, b) -> a.toString().compareTo(b.toString()))
.toArray(Link[]::new);
```

Reply:
I like your suggestion.  I'll fix it.